netkan/netkan/status.py

```diff
@@ -79,50 +81,64 @@ def export_all_mods(cls, compat: bool = True) -> Dict[str, Any]:
+            update_key = f'last_{item_key}'
+            if not values[update_key]:
+                continue
+            values[update_key] = parse(
+                values.pop(update_key)
+            )
+        if values['release_date'] is not None:
+            values['release_date'] = parse(values.pop('release_date'))
+
+        values['ModIdentifier'] = identifier
+        values['success'] = not values['failed']
+        # Assist in migration of tables to multi-game
+        if values['game_id'] is None:
+            values['game_id'] = 'ksp'
+        values.pop('failed')
+
     # This likely isn't super effecient, but we really should only have to use
-    # this operation once to seed the existing history.
+    # this operation once to seed/migrate the existing history.
     @classmethod
     def restore_status(cls, filename: str) -> None:
         existing = json.loads(Path(filename).read_text(encoding='UTF-8'))
         with cls.batch_write() as batch:
-            for key, item in existing.items():
-                # Ensure all date/time items are passed to ModStatus
-                # as proper datetime.datetime obects
-                for item_key in ['checked', 'indexed', 'inflated']:
-                    update_key = f'last_{item_key}'
-                    if not item[update_key]:
-                        continue
-                    item[update_key] = parse(
-                        item.pop(update_key)
-                    )
-                if item['release_date'] is not None:
-                    item['release_date'] = parse(item.pop('release_date'))
-
-                item['ModIdentifier'] = key
-                item['success'] = not item['failed']
-                # Assist in migration of tables to multi-game
-                if item['game_id'] is None:
-                    item['game_id'] = 'ksp'
-                item.pop('failed')
-
-                # Every batch write consumes a credit, we want to leave spare
-                # credits available for other operations and also not error out
-                # during the operation (pynamodb doesn't seem to have a limit
-                # option on batch queries).
-                if len(batch.pending_operations) == 5:
-                    batch.commit()
-                    time.sleep(1)
-
-                batch.save(ModStatus(**item))
+            for game in chain(existing.values()):
+                for key, item in game.items():
+                    # Every batch write consumes a credit, we want to leave spare
+                    # credits available for other operations and also not error out
+                    # during the operation (pynamodb doesn't seem to have a limit
+                    # option on batch queries).
+                    if len(batch.pending_operations) == 5:
+                        batch.commit()
+                        time.sleep(1)
+                    cls.normalise_item(key, item)
```

Comment:
- In the old code, we normalize first and commit batch second
- In the new code, we commit batch first and normalize second

Just wanted to double check that this change is OK and intentional, since it looks like it could be something that happens accidentally in the course of refactoring.

Reply:
It was intentional, probably not super necessary. But realistically this data wasn't getting used until the batch.save() when the status is getting instantiated. It felt more sensible to do it when necessary, rather then before the commit.